app/uk/gov/hmrc/selfservicetimetopay/controllers/CalculatorController.scala

```diff
@@ -34,9 +34,57 @@ class CalculatorController(calculatorConnector: CalculatorConnector) extends Tim
     Redirect(routes.CalculatorController.getAmountsDue())
   }
 
+  def getDebitDate: Action[AnyContent] = Action.async { implicit request =>
+    sessionCache.get.map[Result] {
+      case Some(TTPSubmission(_, _, _, _, Some(EligibilityTypeOfTax(true, false)), Some(EligibilityExistingTTP(Some(false))), _, _, _, _)) =>
+        val dataForm = CalculatorForm.createDebitDateForm()
+        Ok(what_you_owe_date(dataForm))
+      case _ => Redirect(routes.SelfServiceTimeToPayController.start())
+    }
+  }
+
+  def submitDebitDate: Action[AnyContent] = Action.async { implicit request =>
+    sessionCache.get.map[Result] {
+      case Some(ttpData@TTPSubmission(_, _, _, _, _, _, _, _, _, _)) =>
```

Comment:
Do a check on whether user is valid otherwise they could postman this? This seems very minor though will discuss further